src/buffer.rs

```diff
@@ -921,6 +921,35 @@ where
         self.get_pixel_mut(x, y).blend(&p)
     }
 
+    fn copy_within(&mut self, from: (u32, u32), to: (u32, u32), width: u32, height: u32) -> bool {
+        let (fx, fy) = from;
+        let (tx, ty) = to;
+        if tx.max(fx) + width > self.width() || ty.max(fy) + height > self.height() {
+            return false;
+        }
+
+        let px_size = <P as Pixel>::CHANNEL_COUNT as usize * std::mem::size_of::<<P as Pixel>::Subpixel>();
+        let width = px_size * width as usize;
+        if from.1 < to.1 {
+            for y in (0..height).rev() {
+                let sy = fy + y;
```

Comment:
For all these index calculations, please use `pixel_indices_unchecked(x, y).{start,end}`. The current implementation has overflow problems due to the cast to `usize` happening too late. Very similar problems were once fixed by introducing the `pixel_indices` methods :)